src/assets/wise5/authoringTool/add-step-button/add-step-button.component.ts

```diff
@@ -23,18 +24,45 @@ export class AddStepButtonComponent {
   ) {}
 
   protected addStepBefore(): void {
-    this.goToAddStepView(this.projectService.getParentGroupId(this.nodeId));
+    const previousNodes = this.projectService.getNodesByToNodeId(this.nodeId);
+    if (previousNodes.length === 0) {
+      this.goToAddStepViewForIn(this.projectService.getParentGroupId(this.nodeId));
+    } else {
+      const previousNodeId: string = previousNodes[0].id;
+      if (this.projectService.isFirstNodeInBranchPath(this.nodeId)) {
+        this.goToAddStepViewForFirstStepInBranchPath(previousNodeId, this.nodeId);
+      } else {
+        this.goToAddStepViewForAfter(previousNodeId);
+      }
+    }
   }
 
-  protected addStepAfter(): void {
-    this.goToAddStepView(this.nodeId);
+  private goToAddStepViewForIn(groupId: string): void {
+    this.router.navigate(['add-node', 'choose-template'], {
+      relativeTo: this.route,
+      state: {
+        target: new AddStepTarget('in', groupId)
+      }
+    });
+  }
+
+  protected goToAddStepViewForAfter(previousNodeId: string): void {
+    this.router.navigate(['add-node', 'choose-template'], {
+      relativeTo: this.route,
+      state: {
+        target: new AddStepTarget('after', previousNodeId)
+      }
+    });
```

Comment:
Do we need to add the target field, or can we just pass in the AddStepTarget as the state? What else do we need to pass in to the next route?
```suggestion
      state:  new AddStepTarget('after', previousNodeId)
```